File: ios/CAStudy/CAStudy/ViewController.m
//
//  ViewController.m
//  CAStudy
//
//  Created by gdeic on 2/7/12.
//  Copyright (c) 2012 __MyCompanyName__. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController
@synthesize toolbar;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)didReceiveMemoryWarning
{
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc that aren't in use.
}

#pragma mark - View lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)viewDidUnload
{
    [self setToolbar:nil];
	[self setFirstView:nil];
	[self setSecondView:nil];
    [super viewDidUnload];
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    // Return YES for supported orientations
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

- (IBAction)animateItemClick:(id)sender {
	[UIView beginAnimations:@"ToggleViews" context:nil];
	self.firstView.alpha = 0.0;
	self.secondView.alpha = 1.0f;
	[UIView setAnimationDuration:1.0];
	[UIView commitAnimations];
}
@end

File: ios/AnimationCs/ViewController.cs
using System;
using System.Drawing;
using MonoTouch.Foundation;
using MonoTouch.UIKit;

namespace AnimationCs
{
	public partial class ViewController : UIViewController
	{
		public ViewController () : base ("ViewController", null)
		{
		}

		public override void DidReceiveMemoryWarning ()
		{
			// Releases the view if it doesn't have a superview.
			base.DidReceiveMemoryWarning ();
			
			// Release any cached data, images, etc that aren't in use.
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			
			// Perform any additional setup after loading the view, typically from a nib.
		}

		public override bool ShouldAutorotateToInterfaceOrientation (UIInterfaceOrientation toInterfaceOrientation)
		{
			// Return true for supported orientations
			return (toInterfaceOrientation != UIInterfaceOrientation.PortraitUpsideDown);
		}

		partial void SwitchBarItemClick (NSObject sender)
		{
			UIView.Animate(
				1.0,
				1.0,
				UIViewAnimationOptions.CurveEaseIn,
				() => {
					this.FirstView.Alpha = 0.0;
					UIView.Animate(
						1.0,
						1.0,
						UIViewAnimationOptions.OverrideInheritedCurve |
						UIViewAnimationOptions.CurveLinear |
						UIViewAnimationOptions.OverrideInheritedDuration |
						UIViewAnimationOptions.Repeat |
						UIViewAnimationOptions.Autoreverse,
						() => {
							UIView.SetAnimationRepeatCount(2.f);
							this.SecondView.Alpha = 0.0;
						},
						null
					);
				},
				null
			);
		}
	}
}

